authentication.go

```diff
@@ -3,15 +3,22 @@ package jira
 import (
 	"encoding/json"
 	"fmt"
+	"github.com/dgrijalva/jwt-go"
 	"io/ioutil"
 	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
 )
 
 const (
 	// HTTP Basic Authentication
 	authTypeBasic = 1
 	// HTTP Session Authentication
 	authTypeSession = 2
+	// Oauth Authentication
+	authTypeOauth = 3
```

Comment:
I'd call out the oauth version in case another oauth1 is added.  For example, `authTypeOauth2`